docs/limitations.rst

```diff
@@ -0,0 +1,177 @@
+    corruption. Running the application through :doc:`VerifyIO <verifyio>` can
+    help determine if any file locking calls are made.
+
+----------
+
+---------------------------
+MPI-IO and HDF5 Limitations
+---------------------------
+
+Synchronization
+***************
+
+Applications that make use of inter-process communication (e.g., MPI) to enforce
+a particular order of potentially conflicting I/O operations from multiple
+processes must properly use synchronization calls (e.g., ``MPI_File_sync()``) to
+avoid possible data corruption. Within UnifyFS, this requires use of the
+:ref:`"sync-barrier-sync" construct <sync-barrier-sync-label>` to ensure proper
+synchronization.
+
+Properly synchronizing includes doing so between any less obvious MPI-I/O calls
+that also imply a write/read. For example, ``MPI_File_set_size()`` and
+``MPI_File_preallocate()`` act as write operations and ``MPI_File_get_size()``
+acts as a read operation. There may be other implied write/read calls as well.
+
+Synchronization Workarounds
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If your application does not adhere to proper syncronization requirements there
+are four workaround options available to still allow UnifyFS intregration.
+
+UnifyFS WRITE_SYNC
+""""""""""""""""""
+
+UnifyFS provided config option that makes UnifyFS act more “POSIX like” by
+forcing a data sync to the server after **every** write operation. Set
+``UNIFYFS_CLIENT_WRITE_SYNC=ON`` to enable this option.
+
+**Cost:** Can cause a significant decrease in write performance as the amount of
+file sync operations that are performed will be far more than necessary.
+
+Manually Add Sync Operations
+""""""""""""""""""""""""""""
+
+Edit the application code and manually add the proper synchronization calls
+everywhere necessary. For example, wherever an ``MPI_File_sync()`` is required,
+the "sync-barrier-sync" construct needs to be used.
+
+.. _sync-barrier-sync-label:
+
+.. code-block:: C
+    :caption: Sync-barrier-sync Construct
+
+    MPI_File_sync() //flush newly written bytes from MPI library to file system
+    MPI_Barrier()   //ensure all ranks have finished the previous sync
+    MPI_File_sync() //invalidate read cache in MPI library
+
+.. Note::
+
+    The "barrier" in "sync-barrier-sync" can be replaced by a send-recv or
+    certain collectives that are guaranteed to be synchronized. See the "Note on
+    the third step" in the `VerifyIO README`_ for more information.
+
+**Cost:** If making edits to the application source code is an option, the
+amount of time and effort required to track down all the places that proper
+synchonization calls are needed can be very labor intensive.
+:doc:`VerifyIO <verifyio>` can help with in this effort.
+
+HDF5 FILE_SYNC
+""""""""""""""
+
+HDF5 provided config option that forces HDF5 to add an ``MPI_File_sync()`` call
+after every collective write operation when needed by the underlying ROMIO
+driver. Set ``HDF5_DO_MPI_FILE_SYNC=1`` to enable this option.
```

Comment:
Is this option public yet? I couldn't look it up.

Reply:
Public, but not upstreamed yet. It's currently sitting in an open PR (https://github.com/HDFGroup/hdf5/pull/1801) from a `unifyfs` branch in the `hpc-io/hdf5` repo (https://github.com/hpc-io/hdf5/tree/unifyfs).

Reply:
Should I just leave this, assuming the PR will be merged by the time we have a 1.0 release? Alternatively, I could comment this out until the PR is merged or add a link to the branch for using this.

Reply:
It's probably fine to leave it. I doubt someone will try to use it right away. But maybe we should add a note that it will only be available in the next HDF5 release (or a devel snapshot).

Reply:
Added a note on this.